### PR TITLE
[Form][TextInput] Add new components

### LINKED ISSUE #137: [Form][TextInput] new components
As a product developer, I want to use text inputs in my product that follow the guidelines set by the CA Style Guide.

## AC
- A Form component provides layout and onSubmit
- A TextInput takes user input
- support types ('date','datetime-local','email','month','number','password','tel','text','time','url','week')
- Riley and Wendy go over the docs

## Considerations
- Other components to be split out as needed
* Link to styleguide: https://ca.invisionapp.com/d/main#/console/11070131/250301098/preview
* Work on this story has been started and is available here https://github.com/mineral-ui/mineral-ui/tree/137-text-input-deux
* other input "types" (radio, checkbox, etc......)

### PR DESCRIPTION
**Pull request by brentertz**
### Description

* Adds new Form (FormField, FormFieldDivider, & FormFieldset) and TextInput components
* FYI: Updates `.babelrc` to exclude `babel-plugin-polished` during test runs as it is breaking Jest coverage reports.

### Motivation and context

closes #137 

### Screenshots, videos, or demo, if appropriate

https://137-text-input-deux--mineral-ui.netlify.com/component-status/

### How to test

1. View the deploy preview
1. View the component status page
1. View the TextInput page
1. View the Form component pages

### Types of changes

- New feature (non-breaking change which adds functionality)

New Form and TextInput components

- Breaking change (fix or feature that would cause existing functionality to change)

Several keys were removed from the theme.  See commit message for details.

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support)
* [x] Automated tests written and passing
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered
* [x] Documentation created or updated
* [x] Tested in [sandbox](https://github.com/facebookincubator/create-react-app#getting-started) if new component or breaking change

### CODE REVIEW

File: src/website/app/demos/TextInput/bestPractices.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    )
+  },
+  {
+    type: 'dont',
+    description: `Use a \`prefix\` or \`suffix\` as a field label.`,
+    example: <TextInput prefix="Name" />
+  },
+  {
+    type: 'do',
+    description: `Use the appropriate type for the expected content. This is
```
Comment (riley) [inline src/website/app/demos/TextInput/bestPractices.js]: per discussion, maybe make "type" into `type`?

File: src/Form/FormFieldset.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  FormFieldsetLegend_color_text: baseTheme.color_text,
+  FormFieldsetLegend_fontSize: baseTheme.fontSize_ui,
+  FormFieldsetLegend_fontWeight: baseTheme.fontWeight_bold,
+
+  ...baseTheme
+});
+
+const Root = createStyledComponent(
+  'fieldset',
+  props => {
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/Form/FormFieldset.js]: I updated this instance and several others to destructure the props.

File: src/website/app/demos/TextInput/bestPractices.js (hunk 1 above)
Comment (brentertz) [reply src/website/app/demos/TextInput/bestPractices.js]: Done.

File: src/website/app/demos/TextInput/examples/controlled.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+      render() {
+        const textInputProps = {
+          ...this.props,
+          onChange: this.handleTextInputChange,
+          value: this.state.value || 'Hello World'
+        };
+
+        return (
+          <form>
```
Comment (coldpour) [inline src/website/app/demos/TextInput/examples/controlled.js]: do we want this to reload the page if the user hits ENTER?

File: src/website/app/demos/TextInput/index.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  {
+    bestPractices,
+    componentTheme,
+    doc,
+    examples,
+    slug: 'text-input',
+    title: 'TextInput',
+    propsComment: (
+      <p>
+        Unlike other components, undocumented properties will be applied to the{' '}
```
Comment (coldpour) [inline src/website/app/demos/TextInput/index.js]: remove {' '}?

File: src/Form/FormFieldDivider.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+export const componentTheme = (baseTheme: Object) => ({
+  FormFieldDivider_borderColor: baseTheme.borderColor,
+  FormFieldDivider_borderWidth: '1px',
+  FormFieldDivider_margin: baseTheme.space_stack_sm,
+
+  ...baseTheme
+});
+
+const Root = createStyledComponent(
+  'div',
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/Form/FormFieldDivider.js]: I wanted to make it an `hr` as well but I think that @kylegach didn't want to for some reason.  @kylegach Do you recall?

File: src/website/app/demos/TextInput/examples/controlled.js (hunk 3 above)
Comment (brentertz) [reply src/website/app/demos/TextInput/examples/controlled.js]: I removed the form element.

File: src/website/app/demos/TextInput/index.js (hunk 4 above)
Comment (brentertz) [reply src/website/app/demos/TextInput/index.js]: I think Prettier adds that in there to ensure the space is maintained.

File: src/Form/FormFieldDivider.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (brentertz) [reply src/Form/FormFieldDivider.js]: The aria role of [separator](https://www.w3.org/WAI/PF/aria/roles#separator) seems sufficient.

Also, the `hr` element does only allows an aria role of "presentation".

Lastly, there may be some default styling to overcome with an `hr` that is not present in `div`.

Basically, it seems like we get all of the semantic value with `role=separator`, without the headache.

File: src/website/app/demos/Form/bestPractices.js [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+accessible manner when the design requires it.`,
+      example: (
+        <FormField label="Search" hideLabel>
+          <TextInput placeholder="Search..." />
+        </FormField>
+      )
+    },
+    {
+      type: 'dont',
+      description: `Don't hide the label without using the \`hideLabel\` prop.
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/website/app/demos/Form/bestPractices.js]: Removing this example, as it is the direct opposite of the previous one.

File: src/Form/FormField.js [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+ * limitations under the License.
+ */
+
+/* @flow */
+import React, { Children, cloneElement, Component } from 'react';
+import { hideVisually } from 'polished';
+import { createStyledComponent, getNormalizedValue, pxToEm } from '../styles';
+import { generateId } from '../utils';
+
+type Props = {
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/Form/FormField.js]: Improved docs on `caption` prop.  Not sure what else to say about `secondaryText`.

**Pull request merged**